### PR TITLE
Client to class

### PR DESCRIPTION
**Pull request by adespawn**
Update Client to use class syntax

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -51,6 +51,7 @@ const warmupLimit = 32;
  *   If using DCAwareRoundRobinPolicy (default), this option is required and only hosts from this data center are
  *   connected to and used in query plans.
  * </p>
+ * [TODO:]
  * @property {String} [keyspace] The logged keyspace for all the connections created within the {@link Client} instance.
  * @property {Object} [credentials] An object containing the username and password for plain-text authentication.
```
Comment (ZuzaOsa) [inline lib/client.js:56]: Missing TODO?

File: lib/client.js [hunk 2]
```diff
@@ -311,65 +380,76 @@ const warmupLimit = 32;
...
  * <p>
  *   Use <code>null</code> to reset the value and use the default on <code>socketOptions.readTimeout</code> .
  * </p>
+ * [TODO: Delete?]
  * @property {String} [graphOptions.source] The graph traversal source name to use in graph queries. Default:
  * <code>'g'</code>.
+ * [TODO: Delete?]
  * @property {Number} [graphOptions.writeConsistency] Overrides the [consistency
  * level]{@link module:types~consistencies} defined in the query options for graph write queries.
+ * [TODO: Delete?]
```
Comment (ZuzaOsa) [inline lib/client.js:442]: What does the last TODO refrence?

File: lib/client.js [hunk 3]
```diff
@@ -311,65 +380,76 @@ const warmupLimit = 32;
...
  */
 
 /**
- * Creates a new instance of {@link Client}.
- * @classdesc
  * Represents a database client that maintains multiple connections to the cluster nodes, providing methods to
  * execute CQL statements.
- * <p>
```
Comment (ZuzaOsa) [inline lib/client.js:366]: How removing the ``<p>`` symbol changes how the documentation looks?

File: lib/client.js [hunk 4]
```diff
@@ -382,872 +462,865 @@ const warmupLimit = 32;
...
+        return promiseUtils.optionalCallback(this._connect(), callback);
     }
 
-    return promiseUtils.optionalCallback(this._connect(), callback);
-};
+    /**
+     * Async-only version of {@link Client#connect()}.
+     * @private
+     */
+    async _connect() {
```
Comment (ZuzaOsa) [inline lib/client.js:573]: If the _ character is used to denote a private method (I think this was a convention in older JS, but I am not sure) we can check if switching to a [newer syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties) would work.

File: lib/client.js (hunk 2 above)
Comment (adespawn) [reply lib/client.js:442]: It's below property, not above. I can change it if would be more readable.

File: lib/client.js (hunk 3 above)
Comment (adespawn) [reply lib/client.js:366]: In VSC it show the remaining part of the doc as a new paragraph (compared to not showing it at all).
When generating with markdown plugin (``js-dock-export`` branch), it remains the same

File: lib/client.js (hunk 2 above)
Comment (ZuzaOsa) [reply lib/client.js:442]: Oh, I assumed it was above due to the semicolon at the end. I think it is more readable to have it above (``TODO: property`` is more intuitive than ``property TODO``:).

**Comment by ZuzaOsa** [comment]
I think we also want to rebase ``types-support`` and ``simple-query`` on main to add the commit of changing ``new_utils`` to ``new-utils`` (look line 5 of ``client.js``).

**Comment by adespawn** [comment]
> I think we also want to rebase `types-support` and `simple-query` on main to add the commit of changing `new_utils` to `new-utils` (look line 5 of `client.js`).

I forgot that those branches are still not rebased on main.

File: lib/client.js [hunk 5]
```diff
@@ -311,65 +381,75 @@ const warmupLimit = 32;
...
  * </p>
+ * [TODO: Delete?]
  * @property {String} [graphOptions.source] The graph traversal source name to use in graph queries. Default:
  * <code>'g'</code>.
+ * [TODO: Delete?]
  * @property {Number} [graphOptions.writeConsistency] Overrides the [consistency
  * level]{@link module:types~consistencies} defined in the query options for graph write queries.
  */
 
 /**
```
Comment (ZuzaOsa) [inline lib/client.js:445]: When I generated the docs from in-code docs I think this whole upper section describing the parameters is lost (at least for me).

**Comment by ZuzaOsa** [comment]
ClientOptions documentation was moved to a different file. QueryOptions still remains, related issue #81.

**Review by ZuzaOsa — APPROVED** [review]

**Pull request merged**